### PR TITLE
TRUNK-5425: Allow a user to request for a password reset via email

### PR DESCRIPTION
**Pull request by wluyima**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5425

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
    <parent>
       <groupId>org.openmrs</groupId>
       <artifactId>openmrs</artifactId>
-      <version>2.2.0-SNAPSHOT</version>
+      <version>2.2.0-passwordreset</version>
```
Comment (kaweesi) [inline api/pom.xml]: we definitely want to revert the version change except if we wanna maintain this in a separate branch

**Comment by kaweesi** [comment]
i think we need to update this to match changes in master or resolve any possible merge conflicts

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java [hunk 2]
```diff
@@ -597,6 +602,12 @@ public static String getOpenmrsProperty(String property) {
 	 * Specifies the uuid of the concept which represents drug non coded
 	 */
 	public static final String GP_DRUG_ORDER_DRUG_OTHER = "drugOrder.drugOther";
+	
+	/**
+	 * Global property that stores the base url for the application.
+	 */
+	public static final String GP_HOST_URL = "host.url";
```
Comment (kaweesi) [inline api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: do we really need this?

File: api/pom.xml (hunk 1 above)
Comment (wluyima) [reply api/pom.xml]: Yes, good catch!

File: api/src/main/java/org/openmrs/util/OpenmrsConstants.java (hunk 2 above)
Comment (wluyima) [reply api/src/main/java/org/openmrs/util/OpenmrsConstants.java]: How would you know the domain? I think you're assuming it MUST point back to our application which I doubt is always true, an implementation might choose to route it through some other service, they might be behind a proxy etc.

**Comment by dkayiwa** [comment]
@wluyima are we still interested in merging this? Or should we just close it?

**Comment by dkayiwa** [comment]
@fanyui where is your original branch?

**Comment by dkayiwa** [comment]
Merged at: https://github.com/openmrs/openmrs-core/commit/ee6459575982b54fbef0db9db113dc7d10a930f2

**Pull request closed without merging**